src/xdocs/config_imports.xml

```diff
@@ -864,6 +864,65 @@ public class InputIllegalImport { }
+        <p>
+          Short description of the behaviour:
+          <ul>
+            <li>Check starts checking from the longest matching sub-package to the current subpackage described inside import control file.</li>
+            <li>If there is matching allow/disallow rule inside the current subpackage then the Check returns "allowed" or "disallowed" message.</li>
+            <li>If there is no matching allow/disallow rule inside the current subpackage then it continues checking in the parent sub-package.</li>
+            <li>If there is no matching allow/disallow rule in any of the sub-packages, including the root level, then the import is disallowed by default.</li>
+          </ul>
+          <p>To have a <b>blacklist mode</b>, it is required to have disallows inside subpackage and to have allow rule inside parent of the current subpackage to catch classes and packages those are not in the blacklist.</p>
+            <p>In the example below inside subpackage <code>com.puppycrawl.tools.checkstyle.filters</code> any import from <code>java.util</code> (<code>java.util.List</code>, <code>java.util.Date</code>) package except <code>java.util.Map</code> is allowed.</p>
+        </p>
+        <source>
+&lt;import-control pkg=&quot;com.puppycrawl.tools.checkstyle&quot;&gt;
+    &lt;allow pkg=&quot;java.util&quot;/&gt;
+    &lt;subpackage name=&quot;filters&quot; &gt;
+        &lt;disallow class=&quot;java.util.Map&quot;/&gt;
+    &lt;/subpackage&gt;
+&lt;/import-control&gt;
+        </source>
+        <p>
+          In the next example imports <code>java.util.stream.Stream</code> and <code>java.util.stream.Collectors</code> are disallowed inside <code>com.puppycrawl.tools.checkstyle.checks.imports</code> package,
+          but because of <code>&lt;allow pkg=&quot;java.util.stream&quot;/&gt;</code> every import from <code>java.util.stream</code> is allowed except described ones.
+        </p>
+        <source>
+&lt;import-control pkg=&quot;com.puppycrawl.tools.checkstyle&quot;&gt;
+    &lt;subpackage name=&quot;checks&quot;&gt;
+        &lt;allow pkg=&quot;java.util.stream&quot;/&gt;
+        &lt;subpackage name=&quot;imports&quot;&gt;
+            &lt;disallow class=&quot;java.util.stream.Stream&quot;/&gt;
+            &lt;disallow class=&quot;java.util.stream.Collectors&quot;/&gt;
+        &lt;/subpackage&gt;
+    &lt;/subpackage&gt;
+&lt;/import-control&gt;
+        </source>
+
+        <source>
+package com.puppycrawl.tools.checkstyle.checks.imports;
+
+import java.util.stream.Stream;     // violation here
+import java.util.stream.Collectors; // violation here
+import java.util.stream.IntStream;
+        </source>
+        <p>
+          In the following example every imports is allowed except classes <code>java.util.Date</code>, <code>java.util.List</code> and package <code>sun</code>.
+        </p>
+        <source>
+&lt;import-control pkg=&quot;com.puppycrawl.tools.checkstyle&quot;&gt;
+    &lt;subpackage name=&quot;checks&quot;&gt;
```

Comment:
it is allowed only in certain subpackage - "checks". Either change description or remove subpackage "checks".
I recommend to remove subpackage to show example blacklist mode for whole project(allow all is in import-control tag).